ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Analyzers/RedundantLambdaAnalyzer.fs

```diff
@@ -0,0 +1,44 @@
+﻿namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Analyzers
+
+open JetBrains.ReSharper.Feature.Services.Daemon
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings.CommonErrors
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.Util
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl
+
+[<ElementProblemAnalyzer(typeof<ILambdaExpr>, HighlightingTypes = [| typeof<RedundantParenExprWarning> |])>]
+type RedundantLambdaAnalyzer() =
+    inherit ElementProblemAnalyzer<ILambdaExpr>()
+        
+    let rec compareArgs (lambdaArgs: IFSharpPattern seq) (bodyArgs: IFSharpExpression seq) =
+        if bodyArgs.IsEmpty() then lambdaArgs.IsEmpty() else
+        let equal =
+            match (bodyArgs |> Seq.head).IgnoreInnerParens() with
+            | :? ITupleExpr as tuple ->
+                match (lambdaArgs |> Seq.head) with
+                | :? ITuplePat as pat -> compareArgs pat.Patterns tuple.Expressions
+                | :? ILocalAsPat as pat -> compareArgs [pat.Pattern] [tuple]
+                | :? IParenPat as pat -> compareArgs [pat.Pattern] [tuple]
+                | _ -> false
+            | :? IReferenceExpr as reference ->
+                match lambdaArgs |> Seq.head with
+                | :? ILocalReferencePat as pat -> pat.CompiledName = reference.ShortName
+                | :? IParenPat as pat -> compareArgs [pat.Pattern] [reference]
+                | :? ILocalAsPat as pat -> pat.CompiledName = reference.ShortName
+                | _ -> false
+            | _ -> false
+        if equal then compareArgs (Seq.tail lambdaArgs) (Seq.tail bodyArgs) else false
+
+    override x.Run(lambda, _, consumer) =
+        if lambda.Expression == null then () else
```

Comment:
Let's use `isNotNull`? It was added as a sibling to built-in `isNull` (just like it had been done in the compiler).

Using `==` is technically correct here, though, but it was added for cases where we need fast reference equality in PSI instead of F# built-in check that tries to do a lot of extra type checks and conversions in those cases.